R/RLearner_classif_mxff.R

```diff
@@ -0,0 +1,416 @@
+        requires = quote(optimizer == "rmsprop")),
+      makeNumericLearnerParam(id = "gamma2", default = 0.9,
+        requires = quote(optimizer == "rmsprop")),
+      makeNumericLearnerParam(id = "momentum", default = 0, requires = quote(optimizer == "sgd"))
+    ),
+    properties = c("twoclass", "multiclass", "numerics", "prob"),
+    par.vals = list(learning.rate = 0.1, array.layout = "rowmajor", verbose = FALSE),
+    name = "Feedforward Neural Network",
+    short.name = "mxff",
+    note = "Default of `learning.rate` set to `0.1`. Default of `array.layout` set to `'rowmajor'`.
+    Default of `verbose` is set to `FALSE`. If `symbol` is specified, it will be passed to mxnet
+    ignoring other architectural specifications. Default of `initializer` is set to NULL, which
+    results in the default mxnet initializer being called when training a model. Number of output
+    nodes is detected automatically. The upper bound for dropout is set to `1 - 1e-7` as in `mx.mlp`
+    in the `mxnet` package. Dropout should be a numeric vector of length `1`, `2`, or `layers + 1`.
+    If `length(dropout)` is `1`, the same dropout rate will be applied to the inputs and all the
+    hidden layers. If `length(dropout)` is `2`, the first rate will be applied to the inputs, the
+    second to all the hidden layers. If `length(dropout)` is `layers + 1` then the input and every
+    hidden layers will be assigned an individual dropout rate.
+    If `conv.layer1` is `FALSE`, the first layer is a `FullyConnected` layer and `num.layer1` gives
+    the number of neurons. If `conv.layer1` is `TRUE`, then `num.layer1` gives the number of
+    filters. In this case, `act1` is applied as an `Activation` layer afterwards (as is the case
+    with a `FullyConnected` layer).
+    This is the same for `conv.layer2` and `conv.layer3`. A `Convolution`
+    layer cannot follow a `FullyConnected` layer. To stick with the example of the first layer,
+    `c(conv.kernel11, conv.kernel12)`, `c(conv.stride11, conv.stride12)`,
+    `c(conv.dilate11, conv.dilate12)` and `c(conv.pad11, conv.pad12)` correspond to the parameters
+    of `mx.symbol.Convolution`. When a `Convolution` layer is constructed, a `Pooling` layer is
+    constructed with it automatically. Again sticking to the example of the first layer,
+    `c(pool.kernel11, pool.kernel12)`, `c(pool.stride11, pool.stride12)`, `c(pool.pad11, pool.pad12)`
+    and `c(pool.type11, pool.type12)` correspond to the parameters in `mx.symbol.Pooling`.
+    If only the first parameter of a pair is defined, the second parameter is set to the same value,
+    e.g. if `conv.kernel11` is defined and `conv.kernel12` is undefined, `conv.kernel12` is set to
+    the value of `conv.kernel11`.
+    When convolution is used, `conv.data.shape` needs to be specified, which is a vector giving the
+    dimensionality of the data (e.g. for MNIST `c(28, 28)`). Furthermore, `array.layout` is set to
```

Comment:
Does this also allow for three-dimensional inputs, e.g. images with color levels?